File: social-app-backend/src/main/java/razepl/dev/socialappbackend/home/data/PostData.java
package razepl.dev.socialappbackend.home.data;

import lombok.Builder;
import razepl.dev.socialappbackend.globals.EntityData;

import java.time.LocalDate;

/**
 * This record represents post data, including the post content, author, date, username,
 * number of likes, number of comments, post ID, whether the post is liked,
 * and whether the user is in the author's friend list.
 * It implements the EntityData interface.
 */
@Builder
public record PostData(String postContent, String postAuthor, LocalDate postDate, String username,
                       long numOfLikes, long numOfComments, long postId, boolean isPostLiked,
                       boolean isUserInFriends) implements EntityData {
}

File: social-app-frontend/src/app/core/interfaces/home/PostData.ts
/**
 * Represents data about a post.
 */
export interface PostData {
    /**
     * The content of the post.
     */
    postContent: string;

    /**
     * The author of the post.
     */
    postAuthor: string;

    /**
     * The date of the post.
     */
    postDate: Date;

    /**
     * The username associated with the post.
     */
    username: string;

    /**
     * The number of likes on the post.
     */
    numOfLikes: number;

    /**
     * The number of comments on the post.
     */
    numOfComments: number;

    /**
     * The ID of the post.
     */
    postId: number;

    /**
     * Indicates whether the post is liked.
     */
    isPostLiked: boolean;

    /**
     * Indicates whether the user is friends with the author of the post.
     */
    isUserInFriends: boolean;
}
